packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreen.kt

```diff
@@ -0,0 +1,83 @@
+package expo.modules.splashscreen
+
+import android.app.Activity
+import android.util.Log
+import org.unimodules.core.interfaces.SingletonModule
+import java.util.*
+
+object SplashScreen: SingletonModule {
+  private const val TAG = "SplashScreen"
+
+  override fun getName(): String {
+    return "SplashScreen"
+  }
+
+  private val controllers = WeakHashMap<Activity, SplashScreenController>()
+
+  /**
+   * Show SplashScreen by mounting it in ContentView.
+   * @param activity                 Target Activity for SplashScreen to be mounted in.
+   * @param resizeMode               SplashScreen imageView resizeMode.
+   * @param rootViewClass            Class of View that would be monitored for children occurence (autohiding feature).
+   * @param splashScreenConfigurator
+   * @param successCallback          Callback to be called once SplashScreen is mounted in view hierarchy.
+   * @param failureCallback          Callback to be called once SplashScreen cannot be mounted.
+   * @throws [NoContentViewException] when [SplashScreen.show] is called before [Activity.setContentView] (when no ContentView is present for given activity).
+   */
+  @JvmStatic
+  @JvmOverloads
+  fun show(
+    activity: Activity,
+    resizeMode: SplashScreenImageResizeMode,
+    rootViewClass: Class<*>,
+    splashScreenConfigurator: SplashScreenConfigurator = ResourcesBasedSplashScreenConfigurator(),
+    successCallback: () -> Unit = {},
+    failureCallback: (reason: String) -> Unit = { Log.w(TAG, it) }
+  ) {
+    // SplashScreen.show can only be called once per activity
+    if (controllers.containsKey(activity)) {
+      return failureCallback("'SplashScreen.show' has already been called for this activity.")
+    }
+
+    controllers[activity] = SplashScreenController(activity, resizeMode, rootViewClass, splashScreenConfigurator)
+    controllers[activity]!!.showSplashScreen(successCallback)
```

Comment:
`!!` in languages with built-in optionals is something that I really dislike and discourage. It introduces bad coding style. I know this will never throw an exception here, but is it really worth it instead of just adding
```
val controller = SplashScreenController(activity, resizeMode, rootViewClass, splashScreenConfigurator)
```
two lines before?

Also, let's imagine other cases, what if `SplashScreenController` somehow has access to `controllers` and it removes that object from there?